src/rp2_common/cyw43_driver/cyw43_bus_pio_spi.c

```diff
@@ -299,13 +299,13 @@ int cyw43_spi_transfer(cyw43_int_t *self, const uint8_t *tx, size_t tx_length, u
 
         dma_channel_config out_config = dma_channel_get_default_config(bus_data->dma_out);
         channel_config_set_bswap(&out_config, true);
-        channel_config_set_dreq(&out_config, pio_get_dreq(bus_data->pio, 0, true));
+        channel_config_set_dreq(&out_config, pio_get_dreq(bus_data->pio, bus_data->pio_sm, true));
 
-        dma_channel_configure(bus_data->dma_out, &out_config, &bus_data->pio->txf[0], tx, tx_length / 4, true);
+        dma_channel_configure(bus_data->dma_out, &out_config, &bus_data->pio->txf[bus_data->pio_sm], tx, tx_length / 4, true);
 
-        bus_data->pio->fdebug = 1u << PIO_FDEBUG_TXSTALL_LSB;
-        pio_sm_set_enabled(bus_data->pio, 0, true);
-        while (!(bus_data->pio->fdebug & (1u << PIO_FDEBUG_TXSTALL_LSB))) {
+        bus_data->pio->fdebug = 1u << (PIO_FDEBUG_TXSTALL_LSB + bus_data->pio_sm);
+        pio_sm_set_enabled(bus_data->pio, bus_data->pio_sm, true);
+        while (!(bus_data->pio->fdebug & (1u << (PIO_FDEBUG_TXSTALL_LSB + bus_data->pio_sm)))) {
```

Comment:
Does `(1u << (PIO_FDEBUG_TXSTALL_LSB + bus_data->pio_sm))` get collapsed into a compile-time constant, or does it get recalculated for every iteration round the loop?